scripts/apps/monitoring/controllers/AggregateCtrl.js

```diff
@@ -99,10 +102,42 @@ export function AggregateCtrl($scope, api, desks, workspaces, preferencesService
                                 return {'type': 'workspace', 'groups': []};
                             });
                     } else if (activeWorkspace.type === 'desk') {
-                        var desk = self.deskLookup[activeWorkspace.id];
-                        if (desk && desk.monitoring_settings) {
-                            return {'type': 'desk', 'groups': desk.monitoring_settings};
-                        }
+                        return preferencesService.get(PREFERENCES_KEY)
+                            .then(function(preference) {
+                                var desk = self.deskLookup[activeWorkspace.id];
+
+                                // Read available groups from user preferences first
+                                if (preference && preference[activeWorkspace.id] && preference[activeWorkspace.id].groups) {
+                                    if (desk.monitoring_settings !== undefined) {
+                                        // difference between desk.monitoring_settings and preference[activeWorkspace.id].groups
+                                        var diff = _.xor(_.map(desk.monitoring_settings, '_id'),
```

Comment:
I think you can also do:
```js
let monitoringSettings = desk.monitoring_settings || [];
let activePrefGroups = preference[activeWorkspace.id].groups || [];
let ids = _.uniqBy(monitoringSettings.concat(activePrefGroups), '_id');
```
It should have the same outcome and is more obvious imo.